x-pack/plugins/index_management/server/routes/api/inference_models/register_put_inference_route.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema } from '@kbn/config-schema';
+import {
+  InferenceTaskType,
+  InferenceModelConfig,
+  InferencePutModelResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+import { RouteDependencies } from '../../../types';
+import { addBasePath } from '..';
+const paramsSchema = schema.object({
+  taskType: schema.string(),
+  inferenceId: schema.string(),
+});
+export function registerAddInferenceModel({ router, lib: { handleEsError } }: RouteDependencies) {
```

Comment:
I think this makes more sense living in the ML plugin rather than the index management plugin